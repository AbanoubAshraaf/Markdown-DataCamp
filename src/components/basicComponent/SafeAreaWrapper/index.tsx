import React from 'react';
import { SafeAreaView, StyleSheet } from 'react-native';

const SafeAreaWrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return <SafeAreaView style={styles.safeArea}>{children}</SafeAreaView>;
};

const styles = StyleSheet.create({
  safeArea: {
    flex: 1,
  },
});

export default SafeAreaWrapper;
