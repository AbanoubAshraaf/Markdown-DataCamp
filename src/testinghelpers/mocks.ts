export const mockedMarkDownText = 'markdown text'

export const mockedPortraitDimensions = {
    width: 400,
    height: 800,
    scale: 0,
    fontScale: 0,
};

export const mockedLandScapeDimensions = {
    width: 800,
    height: 400,
    scale: 0,
    fontScale: 0,
};