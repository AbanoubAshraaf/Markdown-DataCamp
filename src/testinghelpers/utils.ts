import { RenderResult } from "@testing-library/react-native";

interface Screen extends RenderResult {
    isDetached?: boolean;
}

export const createSnapShot = (screen: Screen) => {
    expect(screen).toMatchSnapshot();
  };