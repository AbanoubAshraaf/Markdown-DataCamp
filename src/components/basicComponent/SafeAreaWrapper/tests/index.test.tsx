import React from 'react';
import { render, screen } from '@testing-library/react-native';
import SafeAreaWrapper from '..';
import { Text } from 'react-native';
import { createSnapShot } from 'src/testinghelpers/utils';

describe('SafeAreaWrapper', () => {
  it('renders children correctly', () => {
    const { getByText } = render(
      <SafeAreaWrapper>
        <Text>Test Child</Text>
      </SafeAreaWrapper>
    );

    expect(getByText('Test Child')).toBeTruthy();
    createSnapShot(screen)
  });
});
