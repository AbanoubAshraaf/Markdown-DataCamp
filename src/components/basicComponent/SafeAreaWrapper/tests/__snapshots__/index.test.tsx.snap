// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SafeAreaWrapper renders children correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <Text>
    Test Child
  </Text>
</RCTSafeAreaView>
`;
